lib/route/neigh.c

```diff
@@ -777,8 +777,15 @@ static int build_neigh_msg(struct rtnl_neigh *tmpl, int cmd, int flags,
 	if (tmpl->ce_mask & NEIGH_ATTR_NHID)
 		NLA_PUT_U32(msg, NDA_NH_ID, tmpl->n_nhid);
 
-	if (tmpl->ce_mask & NEIGH_ATTR_EXT_FLAGS)
-		NLA_PUT_U32(msg, NDA_FLAGS_EXT, tmpl->n_ext_flags);
+	if (tmpl->ce_mask & NEIGH_ATTR_EXT_FLAGS) {
+		/* The kernel does not allow setting the locked flag from
+		 * userspace, so unset it in the request. */
+		uint32_t ext_flags = tmpl->n_ext_flags &
+				     ~(uint32_t)NTF_EXT_LOCKED;
+
+		if (ext_flags)
+			NLA_PUT_U32(msg, NDA_FLAGS_EXT, tmpl->n_ext_flags);
```

Comment:
`NLA_PUT_U32(msg, NDA_FLAGS_EXT, ext_flags);`

Also, could you please squash the commits together? The follow-up commits merely fix the first commit, just rebase (force-push) the pull request.

Reply:
Oops, missed that, good catch.

And yeah I intended to squash once everything was figured out. I'll do that now.